From 6be94d9af0dfd181e334fb21c8c67ade77714966 Mon Sep 17 00:00:00 2001
From: xianbing Zhu <xianbing.zhu@linux.alibaba.com>
Date: Sun, 1 Sep 2024 01:34:03 +0800
Subject: [PATCH 258/400] driver: pty/clk: bootargs options for pty and
 clk_debug init

For reduce time in hibernate resume bootup

Signed-off-by: xianbing Zhu <xianbing.zhu@linux.alibaba.com>
Signed-off-by: Han Gao <gaohan@iscas.ac.cn>
---
 drivers/clk/clk.c | 12 ++++++++++++
 drivers/tty/pty.c | 12 ++++++++++++
 2 files changed, 24 insertions(+)

diff --git a/drivers/clk/clk.c b/drivers/clk/clk.c
index 8474099e2cac..236e2dfa9a24 100644
--- a/drivers/clk/clk.c
+++ b/drivers/clk/clk.c
@@ -3741,6 +3741,16 @@ static void clk_debug_unregister(struct clk_core *core)
 	core->dentry = NULL;
 	mutex_unlock(&clk_debug_lock);
 }
+static bool noclk_debug_init = false;
+
+/* noclkdebug bootargs: for option not init ftrace*/
+static int __init noclk_debug_setup(char *str)
+{
+	noclk_debug_init = true;
+	return 1;
+}
+
+__setup("noclkdebug", noclk_debug_setup);
 
 /**
  * clk_debug_init - lazily populate the debugfs clk directory
@@ -3755,6 +3765,8 @@ static int __init clk_debug_init(void)
 {
 	struct clk_core *core;
 
+	if(noclk_debug_init)
+		return 0;
 #ifdef CLOCK_ALLOW_WRITE_DEBUGFS
 	pr_warn("\n");
 	pr_warn("********************************************************************\n");
diff --git a/drivers/tty/pty.c b/drivers/tty/pty.c
index df08f13052ff..28e0d8b85a82 100644
--- a/drivers/tty/pty.c
+++ b/drivers/tty/pty.c
@@ -937,9 +937,21 @@ static void __init unix98_pty_init(void)
 #else
 static inline void unix98_pty_init(void) { }
 #endif
+static bool nopty = false;
+
+/* noclkdebug bootargs: for option not init ftrace*/
+static int __init nopty_setup(char *str)
+{
+	nopty = true;
+	return 1;
+}
+
+__setup("nopty", nopty_setup);
 
 static int __init pty_init(void)
 {
+	if(nopty)
+		return 0;
 	legacy_pty_init();
 	unix98_pty_init();
 	return 0;
-- 
2.43.0

